src/main/java/org/apache/mesos/state/SchedulerState.java

```diff
@@ -0,0 +1,65 @@
+package org.apache.mesos.state;
+
+import org.apache.mesos.Protos.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Read/write interface for the state of a Scheduler.
+ */
+public class SchedulerState  {
+    private static final Logger log = LoggerFactory.getLogger(SchedulerState.class);
+
+    private static final String SUPPRESSED_KEY = "suppressed";
+    private final StateStore stateStore;
+
+    public SchedulerState(StateStore stateStore) {
+        this.stateStore = stateStore;
+    }
+
+    public Optional<FrameworkID> getFrameworkId() {
```

Comment:
I will, sure.  

But how I really think this should be designed (possibly a future change) is:
- StateStore should be a lower level interface that just deals with key/value storage.
- SchedulerState should have all the scheduler specific state (frameworkId, suppressed, task state, etc.)

As it is, some scheduler state is exposed in StateStore, and some is exposed in SchedulerState (or equivalently, KafkaState and CassandraState), and the distinction seems arbitrary.

Thoughts?